client/src/js/directives/bhLocationSelect.js

```diff
@@ -77,32 +78,29 @@ function LocationSelectController(Locations, $scope) {
 
   /**
    * <select> component messages to be translated
-   * if there is no data, indicate that to the user.
    */
-  var selectCountry  = 'SELECT.COUNTRY';
-  var selectProvince = 'SELECT.PROVINCE';
-  var selectSector   = 'SELECT.SECTOR';
-  var selectVillage  = 'SELECT.VILLAGE';
-  var noData         = 'SELECT.EMPTY';
   vm.messages = {
-    country:  selectCountry,
-    province: selectVillage,
-    sector:   selectSector,
-    village:  selectVillage,
+    country:  Locations.messages.country,
+    province: Locations.messages.province,
+    sector:   Locations.messages.sector,
+    village:  Locations.messages.village,
   };
 
-  // load the countries once, at startup
-  Locations.countries()
-  .then(function (countries) {
-    vm.countries = countries;
-
-    // if there are countries to select, show a "select a country" message
-    // however, if there isn't any data, show a "no data" message. This pattern
-    // is used throughout the component.
-    vm.messages.country = (countries.length > 0) ?
-      selectCountry :
-      noData;
-  });
+  function loadCountries() {
+    Locations.countries()
```

Comment:
I think we discussed this before when discussing previous APIs - in my opinion the angular service abstracts away from the actual HTTP action and does need to directly reflect this in the method name - ```location.countries``` makes it clear what will be returned by this method. 

That being said I can see nothing truly wrong with either suggestion - we should choose one and try to use a similar naming strategy throughout the application.